staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -1083,7 +1083,9 @@ func finishRequest(timeout time.Duration, fn resultFunc) (result runtime.Object,
 	select {
 	case result = <-ch:
 		if status, ok := result.(*metav1.Status); ok {
-			return nil, errors.FromObject(status)
+			if status.Status != metav1.StatusSuccess {
```

Comment:
This is an API change. We will now return a metav1.Status object where we were returning a StatusError object earlier. We should have always been returning a StatusError only in case of errors.
This was the only place where errors.FromObject was being used even when status is StatusSuccess: https://github.com/kubernetes/kubernetes/search?q=errors.FromObject